lib/administrate/field/has_one.rb

```diff
@@ -3,17 +3,25 @@
 module Administrate
   module Field
     class HasOne < Associative
-      def self.permitted_attribute(attr, options = nil)
-        associated_class_name =
-          if options
-            options.fetch(:class_name, attr.to_s.singularize.camelcase)
+      def self.permitted_attribute(attr, options = {})
+        resource_class = options[:resource_class]
+        _associated_class_name =
+          if options.key?(:class_name)
+            Administrate.warn_of_deprecated_option(:class_name)
+            options.fetch(:class_name)
+          elsif resource_class
+            associated_class_name(resource_class, attr)
           else
-            attr
+            Administrate.warn_of_missing_resource_class
+            if options
```

Comment:
do you need to include this `if options` any more?
seems like it was originally included as a check before running `options.fetch` on the next line,
but since you're no longer digging into options it seems useless here.

i can't think of a reason you'd want a branch between `attr.to_s.singularize.camelcase` and plain old `attr`,
but there could be an example i'm missing.

Reply:
I think this is borderline-use enough that I doubt a change will mess up peoples' implementations.
I say scrap our old code, which seems unreasonable.

If you need to branch on the `options` hash, use `if options && options[:class_name]`, to be sure that we're not seeing unintended behavior when someone includes alternative options.

Reply:
Ok, I'm :+1: on your code. Good as is, no breaking changes.